frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,235 @@
+  const isWindows = podToDebug?.spec?.tolerations?.some(
+    (t) => t.key === 'os' && t.value === 'Windows',
+  );
+  return {
+    kind: 'Pod',
+    apiVersion: 'v1',
+    metadata: {
+      name: debugName,
+      namespace: podToDebug?.metadata?.namespace,
+      annotations: {
+        'debug.openshift.io/source-container': containerName,
+        'debug.openshift.io/source-resource': `/v1, Resource=pods/${podToDebug?.metadata?.name}`,
+      },
+    },
+    spec: {
+      activeDeadlineSeconds: 21600,
+      volumes: [],
+      containers: [
+        {
+          name: containerName,
+          image: container?.image,
+          command: isWindows ? ['cmd'] : ['/bin/sh'],
+          resources: {},
+          volumeMounts: [],
+          securityContext: {
+            privileged: true,
+            runAsUser: 0,
+          },
+          stdin: true,
+          stdinOnce: true,
+          tty: true,
+        },
+      ],
+      restartPolicy: 'Never',
+      name: podToDebug?.metadata?.name,
+      hostNetwork: true,
+      hostPID: true,
+      tolerations: podToDebug?.spec?.tolerations || [],
+      nodeSelector: podToDebug?.spec?.nodeSelector || {},
+    },
+  };
+};
+
+const DebugTerminalError: React.FC<DebugTerminalErrorProps> = ({ error }) => {
+  return (
+    <div className="co-m-pane__body">
+      <Alert variant="danger" isInline title={error} />
+    </div>
+  );
+};
+
+const DebugTerminalInner: React.FC<DebugTerminalInnerProps> = ({ obj, activeContainer }) => {
+  const { t } = useTranslation();
+  const infoMessage = (
+    <div className="co-terminal-info-message">
+      <Alert
+        variant="info"
+        isInline
+        title={t(
+          'public~This temporary pod has a modified entrypoint command to debug a failing container. The pod will be available for one hour and will be deleted when the terminal window is closed.',
+        )}
+      />
+    </div>
+  );
+  switch (obj?.data?.status?.phase) {
+    case 'Failed':
+      return (
+        <DebugTerminalError
+          error={
+            <>
+              {t('public~The debug pod failed. ')}
+              {obj?.data?.status?.containerStatuses?.[0]?.state?.terminated?.message ||
+                obj?.data?.status?.message}
+            </>
+          }
+        />
+      );
+    case 'Running':
+      return (
+        <PodExecLoader obj={obj.data} activeContainer={activeContainer} infoMessage={infoMessage} />
+      );
+    default:
+      return <LoadingBox />;
+  }
+};
+
+export const DebugTerminal: React.FC<DebugTerminalProps> = ({
+  obj,
+  match: {
+    params: { name },
+  },
+}) => {
+  const [resources, setResources] = React.useState<FirehoseResource[]>([]);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const podName = obj?.metadata?.name;
+  const podNamespace = obj?.metadata?.namespace;
+  const containerName = name ? name : obj?.spec?.containers?.[0].name;
```

Comment:
These values are required on the pod, so you don't need the extra `?`s

```suggestion
  const podName = obj?.metadata.name;
  const podNamespace = obj?.metadata.namespace;
  const containerName = name || obj?.spec.containers[0].name;
```